js/apps/filter-page/controllers/filterPageCtrl.js

```diff
@@ -30,7 +31,6 @@
 			self.filters = filters;
 			self.allCardData = cardData;
 			self.items = cardData;
-			$scope.$apply();
 
 			angular.element('#results-display').trigger('bcpl.filter.changed', { items: self.items });
```

Comment:
Should you dump this into a method, and call it in setFilter and wherever else you need it. See line 16 and 19